Login and Register Pages Completed (with Oauth2.0 google signin button)

johnskimca opened the pull request:
#### [Title](#)
Login and Register Pages Completed (with Oauth2.0 google signin button)

**Checklist**
- [ ] Run `npm run lint:fix` or `yarn lint:fix`
**Done**

**What and how**
Created the login and sign up pages which includes the component Hyesun made. The pages are forms that have stated which will be connected by redux).
**Screenshots**

![image](https://user-images.githubusercontent.com/59780084/86976941-6973a900-c130-11ea-836d-1967ef849b1b.png)
![image](https://user-images.githubusercontent.com/59780084/86976964-785a5b80-c130-11ea-84ea-8dbefc2eb177.png)

johnskimca committed:
Completed Navbar:

buttonNavBar.js - Created seperate button component (conditional rendering based on props)

johnskimca committed:
Merge branch 'master' of https://github.com/CPSC436/ThinkSmall into navBar

johnskimca committed:
Merge branch 'master' of https://github.com/CPSC436/ThinkSmall into navBar

johnskimca committed:
Completed Navbar:

before pull

johnskimca committed:
Merge branch 'master' of https://github.com/CPSC436/ThinkSmall into navBar

 Conflicts:
	client/src/App.js
	client/src/components/NavBar.js

johnskimca committed:
Fixed issue with API Key -> still having trouble with .env.local

johnskimca committed:
.env.local finished and switch added but cut off

johnskimca committed:
Switch installed, redux set up, and code in map cleaned up

johnskimca committed:
Dispatch is working friend!!

johnskimca committed:
Map vs List View implemented

johnskimca committed:
Pictures added to the infoWindow and resizes based on size of screen.

However, I need help on the following issue: when the screen resizes (goes to small hamburger view or moves out from small screen hamburger view to full screen) the switch resets to list position. It does not remember what switch state it was in before. This means switching screen size can fool our switch to state we are in listview when we are actually in map view.

I'm not sure how I can fix this, unless resizing dispatches an action that is always false (list)

johnskimca committed:
ran:

npm run lint: which caused 5 warnings to appear.
npm run lint:fix (warnings don't appear to be resolved, just errors)

Manually resolved the warning that appears in Switch.js

johnskimca committed:
Update client/src/reducers/index.js

Co-authored-by: Alice <aliceykim0828@gmail.com>

johnskimca committed:
Addressed review notes.

Ran lint and fixed error once more.

Only review note not resolved is in regards to object destructuring.

Please read comment and let me know if this ok

johnskimca committed:
Merge remote-tracking branch 'origin/mapsView' into mapsView

# Conflicts:
#	client/src/reducers/index.js

johnskimca committed:
Merge branch 'project_3' of https://github.com/CPSC436/ThinkSmall into mapsView

 Conflicts:
	client/package.json
	client/src/App.js
	client/src/actions/index.js
	client/src/components/Maps.js
	client/src/components/NavBar/components/Switch.js
	client/src/reducers/index.js

johnskimca committed:
Merge branch 'master' of https://github.com/CPSC436/ThinkSmall into mapsView

johnskimca committed:
Connected redux to maps to render dynamically changing constant.js

johnskimca committed:
Merge branch 'master' of https://github.com/CPSC436/ThinkSmall into optionalMapsFeature

 Conflicts:
	client/src/components/Maps.js

johnskimca committed:
Successfully merged master and Maps Features - ie Maps connected to redux

johnskimca committed:
Created the login and sign up buttons

johnskimca committed:
Merge branch 'master' of https://github.com/CPSC436/ThinkSmall into connectingOauthToButtons

 Conflicts:
	client/src/App.js
	client/src/components/Maps.js
	yarn.lock

johnskimca committed:
Signup Page created through dialog

johnskimca committed:
Form completed with states being saved and validation occuring

johnskimca committed:
Login and Signup completed finished in terms of design

johnskimca committed:
Closes on submit for both login and signup form

johnskimca committed:
Size changed for login dialog

johnskimca committed:
Merge branch 'master' of https://github.com/CPSC436/ThinkSmall into connectingOauthToButtons

 Conflicts:
	client/src/components/Maps.js

johnskimca committed:
Resolved merge conflicts with master and setup db file with .env

alice-0-kim commented on api/db/index.js:
```diff
@@ -1,5 +1,6 @@
 const mongoose = require('mongoose')
 const path = require('path')
+// Committ
```
Bump

yehee commented on api/db/index.js:
```diff
@@ -1,5 +1,6 @@
 const mongoose = require('mongoose')
 const path = require('path')
+// Committ
```
Bump!

johnskimca replied:
Gone!

yehee commented on client/src/components/Form/BusinessForm.js:
```diff
@@ -112,7 +112,7 @@ const Form = ({ open = false, closeForm, addBusiness }) => {
                     maxFileSize={5242880}
                     singleImage
-                    withIcon
+                    withIconz
```
I think this is a typo?

johnskimca replied:
Not sure how that happened but changed!

yehee commented on client/src/components/LoginByGoogle.js:
```diff
@@ -35,15 +26,7 @@ const LoginByGoogle = () => {
-                                    Log Out
-                                </button>
-                            )}
+                            buttonText="Login"
```
Should this be Logout?

johnskimca replied:
Correct, changed!

yehee commented:
LGTM just a minor thing then :shipit:

johnskimca committed:
Resolved review notes in api/db/index.js (comitt)
Typo in business forms,
Finally, LoginByGoogle, signout functinality for some reason has signin text for the button. This has been changed.

johnskimca commented:
Hey Alice, I cleared your review notes but as I pushed to my branch again I think it's asking for another review? Do you want me to just squash and merge as Admin or can you give your ok?
